### ISSUE TITLE
Linux flash tools

### ISSUE THREAD

**Issue by prosper00** [Blinkinlabs/ch554_sdcc#37]
One of the utilities mentioned in the readme - LibreCH551 - appears to be abandoned, and doesn't work with current chips (or, at least, not with current bootloaders on current chips). I'd recommend removing it from the readme, as well as using a different tool in your Makefiles

I've found a handful of alternative utilities that DO work:

VNPro (patched for more CH55x chips) (https://github.com/LoveMHz/vnproch55x)
This is what's used in ch55xduino

ch55x-isptool (https://github.com/ElectronicCats/ch55x-isptool)

chflasher (https://github.com/atc1441/chflasher)

**Comment by ishiyakazuo** [comment]
I've found one more: https://github.com/MarsTechHAN/ch552tool (a Python-based one).  I believe this one also works under Linux (as I seem to recall using it at one point).

**Comment by cibomahto** [comment]
Good point! We're still slowly working our way though a large stockpile of the older chips, which have the reverse problem of not being compatible with the newer programmers.

**Comment by prosper00** [comment]
I think VNPro supports old and new bootloader. At least, the version of VNPro that ch55xduino uses

**Comment by ishiyakazuo** [comment]
LibreCH55x seems to support the oldest bootloaders I've come across.  It's the only one I've seen that supports bootloaders prior to 2.3.0, anyway.
https://github.com/rgwan/librech551/tree/master/usbisp

**Comment by prosper00** [comment]
https://github.com/ole00/chprog claims to support 1.x and 2.x bootloaders

**Comment by jjmz** [comment]
Would you know one that supports writing to the DataFlash area ?

**Comment by cibomahto** [comment]
Finally got back to this project. Using my old parts with a 1.1 bootloader, I tested the following programmers:

* chflasher (python version) detected the part and programmed it successfully
* vnpro detected the part, but dumped out with a 'Not Supported' message.
* ch55xtool (really nice that it's on pypi) unfortunately crashed with an 'operation timed out'
* chprog gives useful status output, and programmed the part successfully
* librech554 also works
